framework_tvb/tvb/core/neocom/_h5loader.py

```diff
@@ -147,6 +147,8 @@ def store(self, datatype):
 
             if self.recursive:
                 sub_dt_refs = f.gather_references()
+                if self.recursive != "deep":  # usually we don't want to store twice the same references
```

Comment:
@dionperd could you pls separate these last commits and put them on a separate branch and PR?
I do not think we can efficiently merge all these together.